design.md

```diff
@@ -0,0 +1,84 @@
+# Bazel Module (bzlmod) Support in Renovate
+
+```yaml
+author: Chuck Grindel<chuck.grindel@gmail.com>
+status: To be reviewed
+created: 2023-04-10
+```
+
+## Abstract
+
+[Renovate] is a popular service for updating external dependencies. It [currently supports] updating
+Bazel dependencies described in [WORKSPACE-managed repositories]. With the introduction of [Bazel
+modules], [Renovate] needs to be modified to support the new external dependency mechanism.
+
+## Requirements
+
+- External Dependency Discovery
+  - Support extracting external dependency information from `MODULE.bazel` files.
+    - Read external dependency information from [bazel_dep] declarations.
+    - Support module overrides.
+      - [Single version override]
+      - [Multiple version override]
+      - [Non-registry overrides]
+- Registry Discovery
+  - Support parsing of `.bazelrc` and files imported in the `.bazelrc` looking for [--registry]
+    flags.
+  - If no registry values are detected, default to the [Bazel Central Registry].
```

Comment:
I think we should also be able to override this in the renovate config for this repository as the dev environment for. a given repo may set these values in some other way.  Same goes for bazel version to support.